compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -426,6 +427,88 @@ static LogicalResult setUserConfig(
   return success();
 }
 
+/// Return the size of the given dimension in the linalg op.
+// TODO: this should be part of LinalgOp interface, the equivalent member
+// function currently only support the case where all the dimensions are static
+// while we want to support dynamic shapes.
+static int64_t getLinalgDimSize(linalg::LinalgOp op, int64_t d) {
```

Comment:
This also seems to work for static shapes. Could you clarify?

Reply:
Could we make the return type `Optional<int64_t>` or `FailureOr<int64_t>` and return `llvm::None` or `failure()` when the dimension is dynamic to not have -1 propagating from here.